src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -277,10 +278,33 @@ def from_neurodata_object(cls, neurodata_object: Container, dataset_name: Litera
             )
             compression_method = "gzip"
         elif dtype == np.dtype("object"):  # Unclear what default chunking/compression should be for compound objects
-            raise NotImplementedError(
-                f"Unable to create a `DatasetIOConfiguration` for the dataset at '{location_in_file}'"
-                f"for neurodata object '{neurodata_object}' of type '{type(neurodata_object)}'!"
-            )
+            # pandas reads in strings as objects by default: https://pandas.pydata.org/docs/user_guide/text.html
+            max_string_length = 1
+            all_elements_are_strings = True
+            for element in candidate_dataset[:].flat:
```

Comment:
You can do this kind of thing with list comprehension which are optimized and I also think is more readable.